includes/class-wc-stripe-feature-flags.php

```diff
@@ -10,6 +10,15 @@ class WC_Stripe_Feature_Flags {
 	 * @return bool
 	 */
 	public static function is_upe_enabled() {
-		return '1' === get_option( '_wcstripe_feature_upe', '0' );
+		return '1' === get_option( '_wcstripe_feature_upe', '0' ) || self::is_upe_settings_redesign_enabled();
```

Comment:
Could this create a weird state in which they disable UPE( `_wcstripe_feature_upe` is 0) but the `_wcstripe_feature_upe_settings` option is still 1 and this returns true?

Reply:
The idea is that `_wcstripe_feature_upe` enables the ability to use UPE in the old settings. And `_wcstripe_feature_upe_settings` enables the new settings (which also have UPE).
So if `_wcstripe_feature_upe` is disabled and `_wcstripe_feature_upe_settings` is enabled, we should indeed show the new settings with the UPE changes.
When the `is_upe_enabled()` method returns true, we need done changes on the gateway classes on both the old and the new settings.
In the end, `_wcstripe_feature_upe=0` and `_wcstripe_feature_upe=1` should be an acceptable state, afaik. It would have the same result as `_wcstripe_feature_upe=1` and `_wcstripe_feature_upe=1`